smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/ClientConfigProperty.kt

```diff
@@ -310,5 +312,33 @@ object KotlinClientRuntimeConfigProperty {
             """.trimIndent()
             propertyType = ClientConfigPropertyType.Required()
         }
+
+        val tracingClientConfigSymbol = buildSymbol {
+            name = "TracingClientConfig"
+            namespace(KotlinDependency.TRACING_CORE)
+        }
+
+        // TODO support a nice DSL for this so that callers don't have to be aware of `DefaultTracer` if they don't want
+        // to, they can just call `SomeClient { tracer { clientName = "Foo" } }` in the simple case.
+        Tracer = ClientConfigProperty {
+            symbol = RuntimeTypes.Tracing.Core.Tracer
+            baseClass = tracingClientConfigSymbol
+            documentation = """
+                The tracer that is responsible for creating trace spans and wiring them up to a tracing backend (e.g.,
+                a trace probe). By default, this will create a standard tracer that uses the service name for the root
+                trace span and delegates to a kotlin-logging trace probe (i.e.,
```

Comment:
nit: I don't know that I'd mention `kotlin-logging` specifically here so much as just mention it will be logged.

Reply:
"Logging" is now an overloaded term because the tracing framework itself "logs" things too. Saying that **kotlin-logging** is involved helps disambiguate at which level the logging will occur and also provides a clue for what to configure to adjust the logging output.

Is the concern about tying to a particular implementation? Providing unnecessary/confusing detail? Other?

Reply:
Mostly exposing what amounts to an implementation detail that we should (are) free to change at will.

Reply:
Okay, will change.